Assets/MRTK/Core/Services/BaseDataProviderAccessCoreSystem.cs

```diff
@@ -62,9 +69,16 @@ public override void LateUpdate()
             {
                 base.LateUpdate();
 
-                foreach (var provider in dataProviders)
+                try
+                {
+                    foreach (var provider in dataProviders)
+                    {
+                        provider.LateUpdate();
```

Comment:
@keveleigh had a really good callout which I didn't see addressed and wanted to mention it specifically - if something fails to get a Update call, should it still get a LateUpdate() call? 

It would probably be better here I think to defer actions on the list while an update loop is occuring, and make it so that if there's a register/unregister it will either get:

1) deferred for register
2) If unregister, we can augment the list with more metadata to indicate that it's pending destruction, so that queries for it don't resolve.

I think otherwise you do risk getting weird behavior where the unregistration of one provider could halt progress for everyone else.

Reply:
I will investigate some more. 

From what I recall when investigating this before, the LateUpdate will not get called due to the shutdown / restore process that occurs. 

I _defintely_ want a cleaner solution here